includes/admin_import.php

```diff
@@ -167,10 +167,40 @@ function tsml_import_page()
 			]);
 			if (is_array($response) && !empty($response['body']) && ($body = json_decode($response['body'], true))) {
 
-				//if already set, hard refresh
+				/*When a data source already exists we want to apply changes detected to the local db */  
+
+				//initialize variables 
+				$feed_updates = $db_ids_to_delete = $message_lines = [];
+				$message = "<br><h2>Local Database Updates Being Applied From the $data_source_name Feed</h2><br>";
+				$data_source_last_import = null;
+
+				//if already set, we want to remove only db records which have been changed or removed from the feed
 				if (array_key_exists($data_source_url, $tsml_data_sources)) {
-					tsml_delete(tsml_get_data_source_ids($data_source_url));
+
+					$tsml_data_sources = get_option('tsml_data_sources', array());
+					$data_source_last_import = (int) $tsml_data_sources[$data_source_url]['last_import'];
```

Comment:
i think `intval` is a more standard way to cast this as an int

Reply:
Legacy code, but I'm willing to learn. What's the advantage of intval over int?

Reply:
as discussed on zoom (but for anyone else looking) i think `intval` is more resilient to unexpected values

Reply:
(int) Typecast / intval()
Typecasting in PHP is a way to force a value into a format such as an integer without condition or recourse. This means that typecasting with (int) is a form of sanitization but not validation and thus does not provide a message when the value passed is not the proper type.

Cons:

It does not validate the input as an integer, it only typecasts it to an integer regardless of passed type.
Malformed IDs are “allowed” in that they are converted to 0 (int) value and the script continues as if you requested an object with an ID of 0. (see php docs warnings below).
Only time (int) typecasting throws an exception is when a class is typecast as an (int) and since WordPress does not inherently catch exceptions for validating it is relatively useless.
The program does not know that a malformed ID was passed to the request until after the database query fails or returns an unexpected result. **early detection was cited as a reason for doing this early and throughout the codebase instead of sanitizing just before post injection, however the implications of an unconditional typecast seem to not have been taken into account.
Pros

Sanitizes input to a integer which is the only ‘type’ an ID can take in the database. Anything other than an integer passed to the database as an ID would be unnecessary.
Faster

Reply:
Done

Reply:
Done